src/panels/lovelace/tile-features/hui-number-tile-feature.ts

```diff
@@ -0,0 +1,122 @@
+import { HassEntity } from "home-assistant-js-websocket";
+import { css, html, LitElement, nothing, PropertyValues } from "lit";
+import { customElement, property, state } from "lit/decorators";
+import { computeDomain } from "../../../common/entity/compute_domain";
+import { isUnavailableState } from "../../../data/entity";
+import { HomeAssistant } from "../../../types";
+import { LovelaceTileFeature, LovelaceTileFeatureEditor } from "../types";
+import { NumberTileFeatureConfig } from "./types";
+import "../../../components/ha-control-button";
+import "../../../components/ha-control-button-group";
+import "../../../components/ha-control-number-buttons";
+import "../../../components/ha-control-slider";
+import "../../../components/ha-icon";
+
+export const supportsNumberTileFeature = (stateObj: HassEntity) => {
+  const domain = computeDomain(stateObj.entity_id);
+  return domain === "input_number" || domain === "number";
+};
+
+@customElement("hui-number-tile-feature")
+class HuiNumberTileFeature extends LitElement implements LovelaceTileFeature {
+  @property({ attribute: false }) public hass?: HomeAssistant;
+
+  @property({ attribute: false }) public stateObj?: HassEntity;
+
+  @state() private _config?: NumberTileFeatureConfig;
+
+  @state() _currentState?: string;
+
+  static getStubConfig(): NumberTileFeatureConfig {
+    return {
+      type: "number",
+      style: "buttons",
+    };
+  }
+
+  public static async getConfigElement(): Promise<LovelaceTileFeatureEditor> {
+    await import("../editor/config-elements/hui-number-tile-feature-editor");
+    return document.createElement("hui-number-tile-feature-editor");
+  }
+
+  public setConfig(config: NumberTileFeatureConfig): void {
+    if (!config) {
+      throw new Error("Invalid configuration");
+    }
+    this._config = config;
+  }
+
+  protected willUpdate(changedProp: PropertyValues): void {
+    super.willUpdate(changedProp);
+    if (changedProp.has("stateObj") && this.stateObj) {
+      this._currentState = this.stateObj.state;
+    }
+  }
+
+  private async _setValue(ev: CustomEvent) {
+    const stateObj = this.stateObj!;
+
+    const domain = computeDomain(stateObj.entity_id);
```

Comment:
can we computeDomain once or memoize it? Now it is calculated every time we want to change the value.

Reply:
This function also only runs on change not on render so it won't run that often.